docs/data/joy/tools/JoyDefaultTheme.js

```diff
@@ -0,0 +1,262 @@
+import * as React from 'react';
+import clsx from 'clsx';
+import PropTypes from 'prop-types';
+import { extendTheme, styled } from '@mui/joy/styles';
+import Box from '@mui/joy/Box';
+import ExpandIcon from '@mui/icons-material/ExpandMore';
+import CollapseIcon from '@mui/icons-material/ChevronRight';
+import TreeView from '@mui/lab/TreeView';
+import MuiTreeItem, { treeItemClasses } from '@mui/lab/TreeItem';
+import { lighten, ThemeProvider, createTheme } from '@mui/material/styles';
+
+/**
+ * @param {unknown} value
+ */
+function getType(value) {
+  if (Array.isArray(value)) {
+    return 'array';
+  }
+
+  if (/^(#|rgb|rgba|hsl|hsla)/.test(value)) {
+    return 'color';
+  }
```

Comment:
Is there a way we could abstract https://github.com/mui/material-ui/blob/defd28e6dc56b136e7af38331c129a3d15006f43/docs/data/material/customization/default-theme/DefaultTheme.js so we don't duplicate the logic?

Reply:
Sure, it can be extracted.